### PR TITLE
COMPAT/TST Matplotlib 2.0 compatability

### LINKED ISSUE #13333: COMPAT: mpl 2.0.0
seems @tacaswell / @mdboom have been busy

https://github.com/matplotlib/matplotlib/releases/tag/v2.0.0b1

we should test with this.

is this going up on a ``conda-forge`` dev channel? (is there such a thing)?

### PR DESCRIPTION
**Pull request by TomAugspurger**
- [x] closes #13333
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Just a WIP for now, I want to cleanup how I handle the errorbar changes.

Will wait for a matplotlib RC before changing the Travis matrix I suppose?

### CODE REVIEW

File: doc/source/whatsnew/v0.19.0.txt [hunk 1]
```diff
@@ -613,3 +613,8 @@ Bug Fixes
 - Bug in ``groupby`` with ``as_index=False`` returns all NaN's when grouping on multiple columns including a categorical one (:issue:`13204`)
 
 - Bug where ``pd.read_gbq()`` could throw ``ImportError: No module named discovery`` as a result of a naming conflict with another python package called apiclient  (:issue:`13454`)
+
+- Compatibility with maptlolib 2.0 (:issue:`13333`)
```
Comment (sinhrks) [inline doc/source/whatsnew/v0.19.0.txt]: As changes are only for tests, there is no bugs related to mpl2. Not sure on what section the description should be...

File: pandas/tests/test_graphics.py [hunk 2]
```diff
@@ -333,6 +336,12 @@ def _check_axes_shape(self, axes, axes_num=None, layout=None,
         figsize : tuple
             expected figsize. default is matplotlib default
         """
+        if figsize is None:
+            if self.mpl_ge_2_0_0:
+                figsize = (6.4, 4.8)
```
Comment (sinhrks) [inline pandas/tests/test_graphics.py]: How about define ``self.default_figsize`` like ``self.polycollection_factor``?

**Comment by TomAugspurger** [comment]
[This build](https://travis-ci.org/pydata/pandas/jobs/145822265) tested against the mpl 2.0 beta. https://travis-ci.org/pydata/pandas/jobs/145822265#L1193 Things seem to be OK.
Changed the release note to

> - Compatibility with maptlolib 2.0. Older versions of pandas should also work with matplotlib 2.0 (:issue:`13333`)

Under "Other enhancements". Look good?

File: pandas/tests/test_graphics.py [hunk 3]
```diff
@@ -346,7 +355,7 @@ def _check_axes_shape(self, axes, axes_num=None, layout=None,
             self.assertEqual(result, layout)
 
         self.assert_numpy_array_equal(
-            np.round(visible_axes[0].figure.get_size_inches()),
+            visible_axes[0].figure.get_size_inches(),  # why round?
```
Comment (jorisvandenbossche) [inline pandas/tests/test_graphics.py]: remove the comment?

Comment (TomAugspurger) [inline pandas/tests/test_graphics.py]: Ahh, right will do. Any idea why we do that though?

Comment (jorisvandenbossche) [inline pandas/tests/test_graphics.py]: no idea

Comment (TomAugspurger) [inline pandas/tests/test_graphics.py]: Forgot I removed it and everything worked.

**Comment by jorisvandenbossche** [comment]
Is this ready? (needs a rebase) Or do you want to wait until matplotlib has a rc release? (but we can also merge, and just update when there is one)

**Comment by TomAugspurger** [comment]
Let me rebase quick and run one of the builds on the mpl pre-release at  `conda install -c conda-forge/label/rc -c conda-forge matplotlib`. In theory pandas 0.18 is already compatible, it was just changes to the tests, so I'm not too worried about the test-suite being run against the RC.

**Comment by jorisvandenbossche** [comment]
There is an error in the py35 build: https://travis-ci.org/pydata/pandas/jobs/152242074, but not sure this is related to this PR

**Comment by TomAugspurger** [comment]
[This](https://travis-ci.org/pydata/pandas/jobs/152442785#L1400) build was the one I made to run against the beta.

As far as I can tell the two failures ([1](https://travis-ci.org/pydata/pandas/jobs/152442781) and [2](https://travis-ci.org/pydata/pandas/jobs/152442783#L1443)) are unrelated, though I don't understand why they failed. Both are linting errors (and one doesn't even look like it actually linted anythnig).

**Comment by TomAugspurger** [comment]
Anyone have a clue what's going on in [here](https://travis-ci.org/pydata/pandas/jobs/153939442#L1401)? The tests pass, but flake8 seems to fail with

> `pkg_resources.DistributionNotFound: The 'configparser' distribution was not found and is required by flake8`

**Comment by jorisvandenbossche** [comment]
I also saw it, I was wondering if adding conda-forge as a channel could have to do something with it? (I don't see any other changes that could cause it). You could check by removing adding the channel and only specifying the channel for the one install line of matplotlib beta.

**Comment by TomAugspurger** [comment]
@jorisvandenbossche that did it 🎉 (no idea why, but oh well).

https://travis-ci.org/pydata/pandas/builds/154140072 was the one run against the MPL 2.0 beta. Good to merge?

**Comment by jorisvandenbossche** [comment]
@TomAugspurger Thanks! 

Maybe it would be good to report that issue to conda-forge somewhere. There is a flake8 feedstock, so possibly something wrong there.

**Pull request merged**